R/FilterStates.R

```diff
@@ -440,12 +440,13 @@ FilterStates <- R6::R6Class( # nolint
             )
           })
 
-
           output$add_filter <- renderUI({
             logger::log_debug(
               "FilterStates$srv_add@1 updating available column choices, dataname: { private$dataname }"
             )
             if (length(avail_column_choices()) == 0) {
+              # because input UI is not rendered on this condition but shiny still holds latest selected value
+              shinyjs::runjs(sprintf("Shiny.setInputValue('%s', null);", session$ns("var_to_add")))
```

Comment:
A method to avoid going through javascript. We use this in `FilterState`.

```suggestion
              .subset2(session$input, "impl")$.values$remove(session$ns("var_to_add"))
```

Reply:
I'm not a fan of this suggestion. I don't really know the difference between `setInputValue(nulll)` and `.$values$remove`. I agree that it is feels safer do interact with shiny socket through the session object than from javascipt.